src/Metal/MTLRenderCommandEncoder.cs

```diff
@@ -10,29 +11,45 @@
 
 namespace Metal {
 	public static class IMTLRenderCommandEncoder_Extensions {
+#if !NET
 		[Mac (10,13), NoiOS, NoTV, NoWatch]
+#else
+		[SupportedOSPlatform ("macos10.13")]
+#endif
 		public unsafe static void SetViewports (this IMTLRenderCommandEncoder This, MTLViewport [] viewports)
 		{
 			fixed (void* handle = viewports)
 				This.SetViewports ((IntPtr)handle, (nuint)(viewports?.Length ?? 0));
 		}
 
+#if !NET
 		[Mac (10,13), NoiOS, NoTV, NoWatch]
+#else
+		[SupportedOSPlatform ("macos10.13")]
+#endif
 		public unsafe static void SetScissorRects (this IMTLRenderCommandEncoder This, MTLScissorRect [] scissorRects)
 		{
 			fixed (void* handle = scissorRects)
 				This.SetScissorRects ((IntPtr)handle, (nuint)(scissorRects?.Length ?? 0));
 		}
 
 #if IOS
+#if !NET
 		[iOS (11,0), NoTV, NoMac, NoWatch]
+#else
+		[SupportedOSPlatform ("ios11.0")]
+#endif
```

Comment:
uho, that file seems to be under a #if MONOMAC so this is not exposed

Reply:
Hmm, should I move the MONOMAC so that these two are exposed?

Reply:
I will expose these in a future PR!

Reply:
Issue created here: https://github.com/xamarin/xamarin-macios/issues/12648